pkg/api/v1/clients/kube/controller/controller.go

```diff
@@ -32,14 +33,15 @@ type Controller struct {
 // Returns a new kubernetes controller without starting it.
 func NewController(
 	controllerName string,
-	handler cache.ResourceEventHandler,
+	handler cache.ResourceEventHandler, RunInformers bool,
```

Comment:
i'd suggest an alternative to passing a bool from various places in the code. there's no indication to clients what `RunInformers` will do, nor that the system will break if more than one client passes `true` here. I think the behavior of ensuring caches are only started once should be deferred to something like a `sync.Once`, or by ensuring that only a single reference to the `Controller` is shared by different clients

Reply:
it not possible as the other object that starts them is the shared informer factory that doesnt use the our controller (as it is kubernetes code).
this patch compiles with gloo - i.e. in gloo the controller is only used via solo-kit and no other change is needed.

Reply:
in the case that a user is using only the `KubeCoreCache` without the CRD `KubeCache`, this approach will fail. for this reason among others i think it is fragile to rely on callers being synchronized without providing any way coordinating the starting of Informers across callers.

wouldn't this be a problem if callers are using different sets of informers and only one set is started? 

why is it not possible ensure informers are only started once via a `sync.Once`? it seems to me that we need a global `Once` for each type of informer we plan to start. the issue gets even more complicated in the case of the apiserver where each token gets its own cache

Reply:
In general global state is not a good idea - It is indeed impossible to have a per token cache with a global `Once`.
I think marco suggestion below should work.

Reply:
i like it